htlcswitch/link.go

```diff
@@ -2868,8 +2782,12 @@ func (l *channelLink) processExitHop(pd *lnwallet.PaymentDescriptor,
 	// Notify the invoiceRegistry of the exit hop htlc. If we crash right
 	// after this, this code will be re-executed after restart. We will
 	// receive back a resolution event.
+	blocksToExpiry := int32(pd.Timeout) - int32(heightNow)
```

Comment:
not sure that converting to `int32` helps here. it's still susceptible to underflow, which would later satisfy either of the equations: `blocksTilExpiry >= finalExpiry` or `blocksTilExpiry >= finalRejectDelta`. however, checking that `pd.Timeout > heightNow` only needs to be on the first accept, which goes in line with my other comment passing both distinctly into `NotifyExitHopHtlc` and validating this in the invoice registry

Reply:
Suppose `pd.Timeout` is `3` and `heightNow` is `5`. `blocksToExpiry` would then be `-2` and `blocksTilExpiry >= finalExpiry` = `-2 >= finalExpiry` would be false?

Reply:
But this check has now indeed been moved to `invoice registry`.

Reply:
The case I'm thinking about is when `pd.Timeout` is very high (as a unit32), like `0x80000000`. In this example, `blocksTilExpiry` is now `2147483643` which exceeds any reasonable `finalExpriy`

Reply:
Rewritten the comparison without subtraction